### ISSUE TITLE
Introduce `#[swift_bridge(Equatable)]` attribute for opaque Rust types

### ISSUE THREAD

**Issue by chinedufn** [chinedufn/swift-bridge#72]
Given the following bridge:

```rust
#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        #[swift_bridge(Equatable)]
        type SomeType;
    }
}

#[derive(PartialEq)]
struct SomeType;
```

We should emit some Swift along the lines of:

```swift
extension SomeTypeRef: Equatable {
    public static func == (lhs: SomeTypeRef, rhs: SomeTypeRef) -> Bool {
        __swift_bridge__$SomeType$_partial_eq(rhs.ptr, lhs.ptr)
    }
}
```

And on the Rust side we'd generate something like:

```rust
#[export_name = "__swift_bridge__$SomeType$_partial_eq"]
pub extern "C" fn __swift_bridge__SomeType__partial_eq (
    lhs: *const super::SomeType,
    rhs: *const super::SomeType
) -> bool {
    unsafe { &*lhs == &*rhs }
}
```

**Comment by chinedufn** [comment]
Here's a guide on how to implement this https://github.com/chinedufn/swift-bridge/issues/77#issuecomment-1381139141

**Comment by chinedufn** [comment]
Closed by #139
